internal/session/session.go

```diff
@@ -142,7 +143,7 @@ func (b *Builder) TimeoutRequestAfter(d time.Duration) session.Builder {
 
 func (b *Builder) Create() (session.Session, error) {
 	var err error
-	if b.connection == nil {
+	if !reflect.ValueOf(b.connection).IsValid() {
```

Comment:
I'm not sure I understand what this is for. Was this to fix functional tests?

Reply:
I think this may have been a fix before upgrading to Go 1.19 or _golangci-lint_ to v1.50, as I have reverted the code and all tests seem to be working.

This is the original error (as `client.Connection` is an interface):
`golang operator == not defined on untyped nil`